src/main/java/org/apache/ibatis/builder/xml/XMLMapperBuilder.java

```diff
@@ -394,9 +395,7 @@ private ResultMapping buildResultMappingFromContext(XNode context, Class<?> resu
   }
 
   private String processNestedResultMappings(XNode context, List<ResultMapping> resultMappings, Class<?> enclosingType) throws Exception {
-    if ("association".equals(context.getName())
-        || "collection".equals(context.getName())
-        || "case".equals(context.getName())) {
+    if (Stream.of("association", "collection", "case").anyMatch(s -> s.equals(context.getName()))) {
```

Comment:
`anyMatch(s -> s.equals(context.getName())` can use the method reference.

```diff
-if (Stream.of("association", "collection", "case").anyMatch(s -> s.equals(context.getName()))) {
+if (Stream.of("association", "collection", "case").anyMatch(context.getName()::equals)) {
```

  And more, above can replace to use `List.contains` (proposal code by IntelliJ IDEA).

```diff
-if (Stream.of("association", "collection", "case").anyMatch(context.getName()::equals)) {
+if (Arrays.asList("association", "collection", "case").contains(context.getName())) {
```

Reply:
@a827871781 @kazuki43zoo I'm making the arrays as list change as it is far cleaner and easier to understand